### DISCUSSION TITLE
Why is Docker not installed on macOS?

### DISCUSSION THREAD

**Discussion by goloroden** [community/community#25777] [Actions]
<p>Hi there :blush:</p>
<p>Currently, macOS does not have Docker installed. Is there a special reason for this, as this is being inconsistent with Linux and Windows?</p>
<p>I have seen <a href="https://github.community/t/is-it-possible-to-install-and-configure-docker-on-macos-runner/16981">https://github.community/t5/GitHub-Actions/Is-it-possible-to-install-and-configure-Docker-on-MacOS-runner/m-p/39002#M3618</a>, but I wonder whether there are official plans to change this?</p>
<p>Have a nice day,</p>
<p>Golo</p>

**Comment by chrispat** [comment]
<p>Docker does not support MacOS it is only avaialbe for Windows and Linux.  The docker desktop developer experience actually runs your containers in a Linux VM and is not something that can be installed on a service as it is not supported or licensed for that scenario.  </p>

**Comment by goloroden** [comment]
<p>First of all, thanks for taking the time to answer :blush:</p>
<p>From a technical point if view, yes, Docker does not run <strong>natively</strong> on macOS.</p>
<p>However, I would like to run some (Node.js) code on macOS, which builds some Docker images. How can I run this on the macOS environment, given that Docker (for Mac) is not installed?</p>
<p>PS: As said in the initial question, I have seen the workarounds… however my question was whether Docker (for Mac) will be officially supported on macOS.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>As mentioned in the linked <a href="https://github.community/t/is-it-possible-to-install-and-configure-docker-on-macos-runner/16981">ticket</a>, docker should be not supported on hosted macOS.</p>
<p>Unfortuately the docker community licensing is such that we are not able to install the docker for mac on our hosted runners.  We have had some discussions with Docker about doing this but they have held firm on their request that using Docker on a service requires an enterprise license and docker enterprise is not at all supported on macOS.</p>
<p>For Windows Microsoft holds the license for Docker Enterprise on Windows so we do not have any issues there and for Linux we use a specific build of Docker for Linux that is built and maintained by Azure.</p>

**Comment by goloroden** [comment]
<p>Thanks for taking the time to explain this!</p>
<p>That’s sad news, especially that it’s something out of your scope (because this means you can’t do anything to change it). I think that this is a pretty dumb decision by Docker, but well… as said, one can’t change it.</p>
<p>Again, thanks for taking the time to reply :slight_smile:</p>

**Comment by reitzig** [comment]
<p>Would it be possible to install another OCI tools, e.g. podman? It’s supposed to be mostly a drop-in replacement, so for many use cases it might be a decent replacement. It could maybe also be used to run Docker actions on macOS.</p>
<p>I mean, macOS sucks for containers, but a <a href="https://developers.redhat.com/blog/2020/02/12/podman-for-macos-sort-of/" rel="nofollow noopener">similar hack as for Docker</a> seems to exist, and I <em>think</em> they have a more liberal stance on licensing; I may be wrong, though.</p>

**Comment by saltzmanjoelh** [comment]
<p>For anyone looking, here is how I got it working for me.</p>
<aside class="onebox githubblob">
  <header class="source">
      <a href="https://github.com/saltzmanjoelh/RunDockerInMacOSAction/blob/a91b09d613f8d0aec2ff50afe9b25a0016618b47/.github/workflows/swift.yml" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <h4><a href="https://github.com/saltzmanjoelh/RunDockerInMacOSAction/blob/a91b09d613f8d0aec2ff50afe9b25a0016618b47/.github/workflows/swift.yml" target="_blank" rel="noopener nofollow ugc">saltzmanjoelh/RunDockerInMacOSAction/blob/a91b09d613f8d0aec2ff50afe9b25a0016618b47/.github/workflows/swift.yml</a></h4>
<pre><code class="lang-yml">name: Swift

on:
  push:
    branches: [main]
  pull_request:
    branches: [main]

jobs:
  
  build-and-test:
    runs-on: macos-latest

    steps:
      - uses: actions/checkout@v2
      - name: Install docker # Taken from https://github.com/actions/virtual-environments/issues/1143#issuecomment-652264388
        run: |
          mkdir -p ~/.docker/machine/cache
          curl -Lo ~/.docker/machine/cache/boot2docker.iso https://github.com/boot2docker/boot2docker/releases/download/v19.03.12/boot2docker.iso
          brew install docker docker-machine
</code></pre>

  This file has been truncated. <a href="https://github.com/saltzmanjoelh/RunDockerInMacOSAction/blob/a91b09d613f8d0aec2ff50afe9b25a0016618b47/.github/workflows/swift.yml" target="_blank" rel="noopener nofollow ugc">show original</a>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by nittysharma** [comment]
<p>Thanks I tried but even after installing docker,  not able to start docker engine.,<br>
can you please check</p>
<p>/usr/local/opt/docker-machine/bin/docker-machine start default<br>
check_suite_focus=true#step:5:61)Creating CA: /Users/runner/.docker/machine/certs/ca.pem</p>
<p>check_suite_focus=true#step:5:62)Creating client certificate: /Users/runner/.docker/machine/certs/cert.pem</p>
<p>check_suite_focus=true#step:5:63)Running pre-create checks…</p>
<p>check_suite_focus=true#step:5:64)Error with pre-create check: “<strong>VBoxManage not found. Make sure VirtualBox is installed and VBoxManage is in the path</strong>”</p>
<p>check_suite_focus=true#step:5:65)Error: Process completed with exit code 3.</p>
